BrainPortal/app/controllers/tasks_controller.rb

```diff
@@ -381,6 +382,15 @@ def create #:nodoc:
       @task.errors[:results_data_provider_id] = 'must be provided'
     end
 
+    if ! @task.bourreau.nil? && ! @task.bourreau.online? # if api mode bourreau is not automatically adjusted
```

Comment:
This is too complicated. Just ad an error and let the rest of the code naturally handle it. No need to add a respons block here when there is already another one.

```ruby
if ! @task.tool_config.bourreau.online?
  @task.errors[:tool_config_id] = 'is on an executions erver that is currently offline'
end
```

Reply:
Yes, that means we get :unprocessable_entity instead of :forbidden but I'm ok with that.